### PR TITLE
Remove deprecated code for administrator

### PR DESCRIPTION
**Pull request by dgrammatiko**
Pull Request for Issue # .
### Summary of Changes
- Move files around to use autoload
- Drop JSubMenuHelper as it was deprecated
- Remove the relative sub menu module
- remove useless entries from administrator/index.php
### Testing Instructions

Admin should still work

### CODE REVIEW

**Comment by zero-24** [comment]
Please add the SQL Update files to remove the module

**Comment by dgt41** [comment]
@zero-24 you mean com_admin/sql ?

**Comment by zero-24** [comment]
Yes :)

**Comment by zero-24** [comment]
As well as the normal install sql under installation/sql

**Comment by dgt41** [comment]
@zero-24 ok, done. Travis happy (at least not complaining about these changes)

**Comment by zero-24** [comment]
@dgt41 i mean the extension ;) As you removed the mod_submenu extension we need to run on update a delete on the extension table and in the install sql we need to remove that extension too :)

**Comment by zero-24** [comment]
:+1:

**Comment by zero-24** [comment]
### Documentation Changes Required

Removed extension & removed methods

File: administrator/components/com_admin/sql/updates/mysql/4.0.0-2016-10-03.sql [hunk 1]
```diff
@@ -0,0 +1,3 @@
+DELETE FROM `#__assets` WHERE `id` = 47;
```
Comment (mbabker) [inline administrator/components/com_admin/sql/updates/mysql/4.0.0-2016-10-03.sql]: This needs to be removed from the update SQL.  We cannot guarantee asset ID 47 is this module.

File: administrator/components/com_admin/sql/updates/mysql/4.0.0-2016-10-03.sql [hunk 2]
```diff
@@ -0,0 +1,3 @@
...
+
+DELETE FROM `#__extensions` WHERE `name` = ''mod_submenu'';
```
Comment (mbabker) [inline administrator/components/com_admin/sql/updates/mysql/4.0.0-2016-10-03.sql]: The quoting on this line is busted.

Comment (wilsonge) [inline administrator/components/com_admin/sql/updates/mysql/4.0.0-2016-10-03.sql]: Fixed that bit with https://github.com/joomla/joomla-cms/blob/4.0-dev/administrator/components/com_admin/sql/updates/mysql/4.0.0-2016-10-03.sql

**Comment by zero-24** [comment]
@wilsonge @dgt41 @mbabker see: https://github.com/joomla/joomla-cms/pull/12281

File: administrator/components/com_admin/sql/updates/mysql/4.0.0-2016-10-03.sql (hunk 1 above)
Comment (mbabker) [inline administrator/components/com_admin/sql/updates/mysql/4.0.0-2016-10-03.sql]: AFAIK we don't have any updates that touch the assets table with direct SQL (aside from schema changes).  It's too risky making changes to its contents for a wide variety of reasons, which is in part why there's that method in the update script to check the base assets for new components are added.

File: administrator/components/com_admin/sql/updates/mysql/4.0.0-2016-10-03.sql (hunk 2 above)
Comment (zero-24) [inline administrator/components/com_admin/sql/updates/mysql/4.0.0-2016-10-03.sql]: hmm this looks not merged see: https://github.com/joomla/joomla-cms/pull/12281/files

Comment (wilsonge) [inline administrator/components/com_admin/sql/updates/mysql/4.0.0-2016-10-03.sql]: I only fixed the quoting with https://github.com/joomla/joomla-cms/commit/2bc5a9a93ace9df8e85d7af66a7165971c15855f - not the other part :)

**Pull request merged**